concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -116,30 +125,122 @@ sectionHeaderLength :: Word64
 sectionHeaderLength = fromIntegral $ BS.length $ encode placeholderSectionHeader
 
 -- |Open a file handle for writing a chunk. If the file with the specified name already exists, a
--- new name is chosen by putting or incrementing a version number in its extension. It is expected
--- that an unversioned filename only has a single extension.
-initialHandle :: FilePath -> IO (FilePath, Handle)
-initialHandle p = do
+-- boolean flag indicates whether the file should be overwritten or whether a new name should be
+-- chosen by adding to or incrementing a version number in its extension. It is expected that an
+-- unversioned filename only has a single extension.
+initialHandle :: Bool -> FilePath -> IO (FilePath, Handle)
+initialHandle overwriteExistingChunk p = do
     createDirectoryIfMissing True (takeDirectory p)
     chunkExists <- doesFileExist p
-    if chunkExists
-        then initialHandle (replaceExtensions p newExt)
+    if chunkExists && not overwriteExistingChunk
+        then do
+            newExt <- getNewExt
+            initialHandle overwriteExistingChunk (replaceExtensions p newExt)
         else do
             hdl <- openBinaryFile p WriteMode
             return (p, hdl)
   where
+    -- The old, potentially unversionsioned, file extension
+    oldExt :: String
     oldExt = takeExtensions p
-    newExt =
+    -- The next versioned file extension
+    getNewExt :: IO String
+    getNewExt =
         if oldExt == takeExtension p
```

Comment:
Since takeExtension and takeExtension*s* are so similar, and the behaviour is very hacky, this warrants documentation.

Reply:
Is the real check here that there is only a single extension? Then that should probably be how it is done, or at least stated.